recipes/zlib-ng/all/conanfile.py

```diff
@@ -0,0 +1,114 @@
+
+    def source(self):
+        get(**self.conan_data["sources"][self.version])
+        os.rename("zlib-ng-{}".format(self.version), self._source_subfolder)
+
+    def _configure_cmake(self):
+        if not self._cmake:
+            self._cmake = CMake(self)
+            self._cmake.definitions["ZLIB_COMPAT"] = False
+            self._cmake.definitions["ZLIB_ENABLE_TESTS"] = False
+            self._cmake.definitions["WITH_GZFILEOP"] = True
+            self._cmake.definitions["WITH_OPTIM"] = True
+            self._cmake.definitions["WITH_NEW_STRATEGIES"] = True
+            self._cmake.definitions["WITH_NATIVE_INSTRUCTIONS"] = False
+            self._cmake.definitions["WITH_SANITIZER"] = False
+            self._cmake.definitions["WITH_FUZZERS"] = False
+            self._cmake.definitions["WITH_MAINTAINER_WARNINGS"] = False
+            self._cmake.definitions["WITH_CODE_COVERAGE"] = False
+
+            self._cmake.configure(build_folder=self._build_subfolder, source_folder=self._source_subfolder)
+        return self._cmake
+
+    def build(self):
+        cmake = self._configure_cmake()
+        cmake.build()
+
+    def package_info(self):
+        self.cpp_info.names["cmake_find_package"] = "ZLIB"
+        self.cpp_info.names["cmake_find_package_multi"] = "ZLIB"
+        if self.settings.os == "Windows":
+            if self.options.shared:
+                if self.settings.build_type == "Debug":
+                    self.cpp_info.libs = ['zlibd']
+                else:
+                    self.cpp_info.libs = ['zlib']
+            else:
+                if self.settings.build_type == "Debug":
+                    self.cpp_info.libs = ['zlibstaticd']
+                else:
+                    self.cpp_info.libs = ['zlibstatic']
+        else:
+            self.cpp_info.libs = ['z-ng']
+
+    def _rename_libraries(self):
+        if self.settings.os == "Windows":
+            lib_path = os.path.join(self.package_folder, "lib")
+            suffix = "d" if self.settings.build_type == "Debug" else ""
+
+            if self.options.shared:
+                if self.settings.compiler == "Visual Studio":
+                    current_lib = os.path.join(lib_path, "zlib%s.lib" % suffix)
+                    os.rename(current_lib, os.path.join(lib_path, "zlib.lib"))
+            else:
+                if self.settings.compiler == "Visual Studio":
+                    current_lib = os.path.join(lib_path, "zlibstatic%s.lib" % suffix)
+                    os.rename(current_lib, os.path.join(lib_path, "zlib.lib"))
+                elif self.settings.compiler == "gcc":
+                    current_lib = os.path.join(lib_path, "libzlibstatic.a")
+                    os.rename(current_lib, os.path.join(lib_path, "libzlib.a"))
+                elif self.settings.compiler == "clang":
+                    current_lib = os.path.join(lib_path, "zlibstatic.lib")
+                    os.rename(current_lib, os.path.join(lib_path, "zlib.lib"))
```

Comment:
libraries are renamed here, but in the meantime you define lib names in package_info() like if they were not renamed.